admin/yum.py

```diff
@@ -0,0 +1,176 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Effectful interface to RPM tools.
+"""
+
+import os
+from urlparse import urlparse
+
+from twisted.python.filepath import FilePath
+from characteristic import attributes
+from effect import Effect, sync_performer, TypeDispatcher
+from effect.do import do
+from subprocess import check_call
+from textwrap import dedent
+
+
+@attributes([
+    "source_repo",
+    "target_path",
+    "packages",
+])
+class DownloadPackagesFromRepository(object):
+    """
+    Download a given set of RPMs from a repository.
+
+    :ivar bytes source_repo: Location of repoisitory.
+    :ivar FilePath target_path: Directory to download packages to.
+    :ivar list packages: List of bytes, package names to download.
+    """
+
+
+@sync_performer
+def perform_download_packages_from_repository(dispatcher, intent):
+    """
+    See :class:`DownloadPackagesFromRepository`.
+    """
+    # TODO don't use yum tools. Just download files.
+    import urllib2
+    import urllib
```

Comment:
I'm a little sad to suggest another dependency, but I think I should say, just use http://docs.python-requests.org/en/latest/